### ISSUE TITLE
DB Upgrade failure from 1.8.1 to 1.9.1 - relation "director_activity_log_remark" does not exist

### ISSUE THREAD

**Issue by sqs-it** [Icinga/icingaweb2-module-director#2596]
## Expected Behavior
Upgrade to conclude without errors

## Current Behavior
In activity log (and other pages) you get the following error:
Oops, an error occurred!
```
SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "director_activity_log_remark" does not exist
LINE 1: SELECT director_activity_log_remark.* FROM director_activity...
                                                   ^, query was: SELECT director_activity_log_remark.* FROM director_activity_log_remark WHERE (first_related_activity <= 856) AND (last_related_activity >= 757) (Pdo.php:225)
```
## Steps to Reproduce (for bugs) (not sure if it is reproducible)
From an 1.8.1 upgrade to 1.9.1 using the release tarball install method 

## Your Environment
* Director version (System - About): 1.9.1
* Icinga Web 2 version and modules (System - About): 2.9.5
* Icinga 2 version (`icinga2 --version`): 2.13.2-1
* Operating System and version: Rocky Linux 8.6
* Webserver, PHP versions: php.x86_64 7.2.24-1, httpd 2.4.37-47

## Extra Info
After an upgrade I can see in the Icinga Director -> Health tab the indication that there are 7 pending schema migrations.
Next to the Activity log text there is a 7 with a light blue background indicating the pending operations, but when I click the activity log I get presented the error above.

I noticed that if I try to access other pages from the Icinga Director ie. Hosts I get the error o.uuid refereed in issue #2593 so this might be related

Remaining output from Activity Log page
```
#0 /usr/share/icingaweb2/library/vendor/Zend/Db/Statement.php(297): Zend_Db_Statement_Pdo->_execute(Array)
#1 /usr/share/icingaweb2/library/vendor/Zend/Db/Adapter/Abstract.php(470): Zend_Db_Statement->execute(Array)
#2 /usr/share/icingaweb2/library/vendor/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('SELECT director...', Array)
#3 /usr/share/icingaweb2/library/vendor/Zend/Db/Adapter/Abstract.php(725): Zend_Db_Adapter_Pdo_Abstract->query(Object(Zend_Db_Select), Array)
#4 /usr/share/icingaweb2/modules/director/library/Director/Web/Table/ActivityLogTable.php(75): Zend_Db_Adapter_Abstract->fetchAll(Object(Zend_Db_Select))
#5 /usr/share/icingaweb2/modules/incubator/vendor/gipfl/icingaweb2/src/Table/QueryBasedTable.php(230): Icinga\Module\Director\Web\Table\ActivityLogTable->fetchQueryRows()
#6 /usr/share/icingaweb2/modules/incubator/vendor/gipfl/icingaweb2/src/Table/QueryBasedTable.php(164): gipfl\IcingaWeb2\Table\QueryBasedTable->fetch()
#7 /usr/share/icingaweb2/modules/incubator/vendor/gipfl/icingaweb2/src/Table/QueryBasedTable.php(115): gipfl\IcingaWeb2\Table\QueryBasedTable->fetchRows()
#8 /usr/share/icinga-php/ipl/vendor/ipl/html/src/BaseHtmlElement.php(323): gipfl\IcingaWeb2\Table\QueryBasedTable->renderContent()
#9 /usr/share/icinga-php/ipl/vendor/ipl/html/src/HtmlDocument.php(498): ipl\Html\BaseHtmlElement->renderUnwrapped()
#10 /usr/share/icinga-php/ipl/vendor/ipl/html/src/HtmlDocument.php(339): ipl\Html\HtmlDocument->render()
#11 /usr/share/icinga-php/ipl/vendor/ipl/html/src/BaseHtmlElement.php(255): ipl\Html\HtmlDocument->renderUnwrapped()
#12 /usr/share/icinga-php/ipl/vendor/ipl/html/src/BaseHtmlElement.php(323): ipl\Html\BaseHtmlElement->renderContent()
#13 /usr/share/icinga-php/ipl/vendor/ipl/html/src/HtmlDocument.php(498): ipl\Html\BaseHtmlElement->renderUnwrapped()
#14 /usr/share/icinga-php/ipl/vendor/ipl/html/src/HtmlDocument.php(369): ipl\Html\HtmlDocument->render()
#15 /usr/share/icingaweb2/modules/incubator/vendor/gipfl/icingaweb2/src/Zf1/SimpleViewRenderer.php(65): ipl\Html\HtmlDocument->__toString()
#16 /usr/share/icingaweb2/modules/incubator/vendor/gipfl/icingaweb2/src/Zf1/SimpleViewRenderer.php(104): gipfl\IcingaWeb2\Zf1\SimpleViewRenderer->render()
#17 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action/HelperBroker.php(272): gipfl\IcingaWeb2\Zf1\SimpleViewRenderer->postDispatch()
#18 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action.php(518): Zend_Controller_Action_HelperBroker->notifyPostDispatch()
#19 /usr/share/php/Icinga/Web/Controller/Dispatcher.php(76): Zend_Controller_Action->dispatch('activitiesActio...')
#20 /usr/share/icingaweb2/library/vendor/Zend/Controller/Front.php(937): Icinga\Web\Controller\Dispatcher->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))
#21 /usr/share/php/Icinga/Application/Web.php(304): Zend_Controller_Front->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))
#22 /usr/share/php/Icinga/Application/webrouter.php(107): Icinga\Application\Web->dispatch()
#23 /usr/share/icingaweb2/public/index.php(4): require_once('/usr/share/php/...')
#24 {main}
```

**Comment by sqs-it** [comment]
This is what I see when I click the activity log menu entry on the left.
Can-t see where to click to apply the migrations.... But I might be missing something obvious... 
![image](https://user-images.githubusercontent.com/112553346/191473908-3e195cbf-9762-4352-ae9d-c13b9b2c2c72.png)

**Comment by Thomas-Gelf** [comment]
I see, this should be fixed. However, please click "Icinga Director" in the menu - it should offer you a possibility to apply the migration

**Comment by sqs-it** [comment]
Ok, I' sorry I did completely missed the apply Migrations button in the main Icinga Director.

When I did it gave me 1 error you can see in the following pic:
![image](https://user-images.githubusercontent.com/112553346/191563251-ae7034f2-0a07-4f5c-8482-60d697d6fc7b.png)


When I refreshed the page it already only said 5 migrations left so I clicked it again and got the following error:
![image](https://user-images.githubusercontent.com/112553346/191563120-80850f3b-5daa-4dd7-aa9f-eb15ca4aa11a.png)

And now will apply any more migrations.

**Comment by idandr** [comment]
Maybe https://github.com/Icinga/icingaweb2-module-director/issues/2482#issuecomment-1050915582 helps? Or https://github.com/Icinga/icingaweb2-module-director/issues/2482#issuecomment-1050931556 in case you run PostgeSQL before 9.4.

**Comment by sqs-it** [comment]
Hello @idandr thanks for the tip, that solved it.
This is a old Icinga install I've been upgrading through the times and somewhere I probably missed a requirement in the release notes. (my bad)

Leaving a few details just in case someone else has a similar problem.

To install the extension 1st connect to the DB as superuser:
`sudo -u postgres psql -d director`

Then run the command:
`CREATE EXTENSION pgcrypto;`

But I was getting the following error:
`ERROR:  could not open extension control file "/usr/share/pgsql/extension/pgcrypto.control": No such file or directory`

So after some digging I found out I was missing the `postgresql-contrib` package that contains the pgcrypto extension, so got to install that:
`sudo dnf install postgresql-contrib`

After that I could verify that the extension was available in the `/usr/share/pgsql/extension/` folder.

So connect to the DB again, `sudo -u postgres psql -d director`
Add the extension do the DB with : `CREATE EXTENSION pgcrypto;`
This time no errors reported.
I confirmed that the extension was available with the following command: 
`director=# SELECT * FROM pg_extension;
 extname  | extowner | extnamespace | extrelocatable | extversion | extconfig | extcondition
----------+----------+--------------+----------------+------------+-----------+--------------
 plpgsql  |       10 |           11 | f              | 1.0        |           |
 pgcrypto |       10 |         2200 | t              | 1.3        |           |
(2 rows)
`
Looks good!
After that went to the icingaweb interface and applied the 5 missing DB migrations, which completed successfully.
My Icinga install is running smoothly again!
